ui/hooks/gasFeeInput/useTransactionFunctions.js

```diff
@@ -87,7 +88,11 @@ export const useTransactionFunctions = ({
         );
         dispatch(updateCustomSwapsEIP1559GasParams(newGasSettings));
       } else {
-        dispatch(updateTransactionFn(updatedTxMeta));
+        const userSettings = {};
+        userSettings.userEditedGasLimit = updatedTxMeta.userEditedGasLimit;
+        userSettings.userFeeLevel = updatedTxMeta.userFeeLevel;
+        dispatch(updateTransactionGasFees(updatedTxMeta.id, newGasSettings));
+        dispatch(updateTransactionUserSettings(updatedTxMeta.id, userSettings));
```

Comment:
`txMeta` may have more fields, do we ensure that all is saved ?

https://github.com/MetaMask/metamask-extension/pull/13646/files#diff-a37b7544a0b876163dca658b96d70b11e99c7d7c33cb64ef2295c2244343b981R73

Reply:
@jpuri The essence of this change is that we don't want to update `all` fields, just the ones that should be updated. In this case the gas settings.

Reply:
@darkwing can't do that because of [this](https://eslint.org/docs/rules/no-case-declarations)

Reply:
@segun : but in this case there are more fields than gas settings in txMeta which will be lost.

Reply:
@jpuri I am updating gas fees and userSettingsFees based on 

1. the parameters passed to `useTransactionFunction`
2. the computed values in the method.

Reply:
@danjm : in EIP-1559 V2 we added this field `previousGas` in txMeta. This is used in case of cancel/speedup and saves the gas used in original transaction. This field is used to derive the new estimate type `10% Minimum` for the transaction.

Commit here [fixes](https://github.com/MetaMask/metamask-extension/pull/13646/commits/83652f84fa8a631e6bcabf8a0af2d4ef9d5159a9#diff-675c437262d906486cf5436c0b64535347286cc4d1081dbd863fb45b8a01b81aR386) this issue.